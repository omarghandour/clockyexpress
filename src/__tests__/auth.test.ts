// src/__tests__/userRoutes.test.ts
import request from "supertest";
import express from "express";
import userRoutes from "../routes/userRoutes";
import User from "../models/user";
import generateToken from "../utils/generateToken";

// Mock the User model and utility functions
jest.mock("../models/user");
jest.mock("../utils/generateToken");

const app = express();
app.use(express.json());
app.use("/api/users", userRoutes);

describe("User Routes", () => {
  it("should register a new user and match snapshot", async () => {
    const newUser = {
      _id: "1",
      name: "John Doe",
      email: "john@example.com",
      password: "password123",
    };

    (User.findOne as jest.Mock).mockResolvedValue(null);
    (User.prototype.save as jest.Mock).mockResolvedValue(newUser);
    (generateToken as jest.Mock).mockReturnValue("fakeToken");

    const res = await request(app)
      .post("/api/users/register")
      .send({
        name: "John Doe",
        email: "john@example.com",
        password: "password123",
      });

    expect(res.status).toBe(201);
    expect(res.body).toMatchSnapshot(); // This will generate and compare the snapshot
  });

  // Add more tests here
});
