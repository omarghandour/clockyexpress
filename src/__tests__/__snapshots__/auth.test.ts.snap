// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Routes should register a new user and match snapshot 1`] = `
{
  "_id": "1",
  "email": "john@example.com",
  "name": "John Doe",
  "token": "fakeToken",
}
`;
